### PR TITLE
Minor: Add `RuntimeEnvBuilder::build_arc()

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?

N/A

## Rationale for this change

Small follow on to https://github.com/apache/datafusion/pull/12157 from @devanbenz 

It is still somewhat awkward to make RuntimeEnv's, so let's make it slightly easier

## What changes are included in this PR?

Adds a convenience method to `RuntimeEnvBuilder` to build an `Arc<RuntimeEnv>`

## Are these changes tested?
Yes, by CI

## Are there any user-facing changes?
A new method

### CODE REVIEW

File: datafusion/core/tests/fuzz_cases/sort_fuzz.rs [hunk 1]
```diff
@@ -136,12 +136,10 @@ impl SortTest {
                     .sort_spill_reservation_bytes,
             );
 
-            let runtime = Arc::new(
-                RuntimeEnvBuilder::new()
-                    .with_memory_pool(Arc::new(GreedyMemoryPool::new(pool_size)))
-                    .build()
-                    .unwrap(),
-            );
+            let runtime = RuntimeEnvBuilder::new()
```
Comment (alamb) [inline datafusion/core/tests/fuzz_cases/sort_fuzz.rs:139]: this is a pretty good example of how the new API makes the code simpler

**Review by findepi — APPROVED** [review]
I like the simplification.
Is this an established naming pattern in the ecosystem?

**Comment by alamb** [comment]
> I like the simplification. Is this an established naming pattern in the ecosystem?

I am not aware of this being an established pattern (I think DataFusion uses Arc's a bit more than many Rust purists would prefer)

**Review by jayzhan211 — APPROVED** [review]
👍

**Comment by jayzhan211** [comment]
I also have the same thought, maybe we don't really need Arc in some cases. We could gradually cleanup them later on

**Comment by jayzhan211** [comment]
Thanks @alamb and @findepi

**Pull request merged**